holidays/countries/guinea.py

```diff
@@ -0,0 +1,183 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT, NOV, DEC
+from holidays.groups import ChristianHolidays, InternationalHolidays, IslamicHolidays
+from holidays.observed_holiday_base import ObservedHolidayBase, SUN_TO_NEXT_MON
+
+
+class Guinea(ObservedHolidayBase, ChristianHolidays, InternationalHolidays, IslamicHolidays):
+    """Guinea holidays.
+
+    References:
+        * [Decree No. 2022-0526](https://igt.gov.gn/wp-content/uploads/2024/10/D-2022-0526-PRG-CNRD_221103_131021.pdf)
+        * <https://www.timeanddate.com/holidays/guinea/>
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Guinea>
+        * <https://anydayguide.com/calendar/1878>
+        * <https://www.timeanddate.com/holidays/guinea/second-republic-day>
+        * <https://www.timeanddate.com/holidays/guinea/all-saints-day>
+
+    Note:
+        According to Decree No. 2022-0526(Article 2) of 2 November 2022, if New Year's Day,
+        Independence Day or Eid al-Fitr fall on a non-working day, the next working day is
+        also a Holiday.
+        Eid al-Adha was assigned two holidays from 2022(Decree No. 2022-0526(Article 1))
```

Comment:
```suggestion
    According to Decree No. 2022-0526 of 2 November 2022:
        * Eid al-Adha became a two-day holiday (Article 1).
        * If New Year's Day, Independence Day or Eid al-Fitr fall on a non-working day,
            the next working day is also a holiday (Article 2).
```
I think this is my last suggestion.